jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -597,4 +667,100 @@ public String toString()
                 getPooled());
         }
     }
+
+    public interface Holder<P>
+    {
+        Pool.Entry<P> getEntry();
+
+        void hold();
```

Comment:
How about renaming this method `harden()`?

Reply:
I prefer `release()` as it matches where it is called from.

Reply:
@sbordet but it is not a release.  A release only reduces the multiplexed reference count and only if that goes to zero do we create a non-weak reference.   So it is called from release, but is not the same.
Maybe `lastRelease` or `fullyReleased` ?   But for now I'll go with `harden` unless there is other concensus.